warehouse/query-core/src/main/java/datawave/query/attributes/UniqueGranularity.java

```diff
@@ -32,6 +32,36 @@ public enum UniqueGranularity {
      */
     TRUNCATE_TEMPORAL_TO_HOUR("HOUR", new DateTimeValueFormatter("yyyy-MM-dd'T'HH")),
     
+    /**
+     * A {@link UniqueGranularity} implementation that, if provided a datetime value, will return the datetime truncated to the month. Otherwise, the original
+     * value will be returned.
+     */
+    TRUNCATE_TEMPORAL_TO_MONTH("MONTH", new DateTimeValueFormatter("yyyy-MM")),
+    
+    /**
+     * A {@link UniqueGranularity} implementation that, if provided a datetime value, will return the datetime truncated to the year. Otherwise, the original
+     * value will be returned.
+     */
+    TRUNCATE_TEMPORAL_TO_YEAR("YEAR", new DateTimeValueFormatter("yyyy")),
+    
+    /**
+     * A {@link UniqueGranularity} implementation that, if provided a datetime value, will return the datetime truncated to the second. Otherwise, the original
+     * value will be returned.
+     */
+    TRUNCATE_TEMPORAL_TO_SECOND("SECOND", new DateTimeValueFormatter("yyyy-MM-dd'T'HH:mm:ss")),
+    
+    /**
+     * A {@link UniqueGranularity} implementation that, if provided a datetime value, will return the datetime truncated to the millisecond. Otherwise, the
+     * original value will be returned.
+     */
+    TRUNCATE_TEMPORAL_TO_MILLISECOND("MILLISECOND", new DateTimeValueFormatter("yyyy-MM-dd'T'HH:mm:ss.SSS")),
+    
+    /**
+     * A {@link UniqueGranularity} implementation that, if provided a datetime value, will return the datetime truncated to the tenth of an hour. Otherwise, the
+     * original value will be returned.
+     */
+    TRUNCATE_TEMPORAL_TO_TENTH("TENTH", new DateTimeValueFormatter("mm", true)),
```

Comment:
Since TENTH isn't a common time unit like the others, I might extend this to TENTH_OF_HOUR for clarity. 

The format specifier appears to just be "minutes". Is that correct?

Reply:
Right, the idea would be to get the tenth digit from the minutes from the date and do an unique on that. I can clear that up.